docs/data/date-pickers/custom-field/SpaciousFieldFormat.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import dayjs from 'dayjs';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateField } from '@mui/x-date-pickers/DateField';
+
+export default function SpaciousFieldFormat() {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <DemoContainer components={['DateField']}>
+        <DateField defaultValue={dayjs('2022-04-17')} formatDensity="spacious" />
```

Comment:
What do you think about showcasing all the options to bring more clarity?
Maybe even list them in the same sense as `am/pm` demos:
- default
- `spacious`
- `dense`